projects/valtimo/process-link/src/lib/components/select-url/select-url.component.ts

```diff
@@ -0,0 +1,194 @@
+                this.updateProcessLink();
+            } else {
+                this.saveNewProcessLink();
+            }
+        });
+    }
+
+    private updateProcessLink(): void {
+        this.stateService.selectedProcessLink$
+            .pipe(take(1))
+            .subscribe((selectedProcessLink) => {
+                const updateProcessLinkRequest: URLProcessLinkUpdateRequestDto = {
+                    id: selectedProcessLink.id,
+                    url: this.url.value,
+                };
+
+                this.processLinkService.updateProcessLink(updateProcessLinkRequest).subscribe(
+                    () => {
+                        this.stateService.closeModal();
+                    },
+                    () => {
+                        this.stateService.stopSaving();
+                    }
+                );
+            });
+    }
+
+    private saveNewProcessLink(): void {
+        combineLatest([
+            this.stateService.modalParams$,
+            this.stateService.selectedProcessLinkTypeId$
+        ])
+            .pipe(
+                take(1),
+                switchMap(([modalParams, processLinkTypeId]) =>
+                    this.processLinkService.saveProcessLink({
+                        url: this.url.value,
+                        activityType: modalParams.element.activityListenerType,
+                        processDefinitionId: modalParams.processDefinitionId,
+                        processLinkType: processLinkTypeId,
+                        activityId: modalParams.element.id
+                    })
+                )
+            )
+            .subscribe(
+                () => {
+                    this.stateService.closeModal();
+                },
+                () => {
+                    this.stateService.stopSaving();
+                }
+            );
+    }
+
+    public resolveUrlVariables(url: string) {
+        return this._variables$.pipe(
+            take(1),
+            map(variables => resolveUrlVariables(url, variables))
+        )
+    }
+
+    get url() {
+        return this.urlForm.get('url');
+    }
+}
+
+export function urlValidator(variables: Observable<Map<string, string>>): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<ValidationErrors | null> => {
+        return variables.pipe(
+            take(1),
+            map(variables => {
+                const url = resolveUrlVariables(control.value, variables)
+                try {
+                    new URL(url);
+                    return null;
+                } catch (_) {
+                    return {invalidUrl: url};
+                }
+            })
+        )
+    }
+}
+
+export function resolveUrlVariables(url: string, variables: Map<string, string>): string {
+    let resolvingUrl = url
+    Object.keys(variables).forEach(key => {
+        let value = variables[key]
+        resolvingUrl = resolvingUrl.replace("<" + key + ">", value)
+    })
+    return resolvingUrl
+}
```

Comment:
If you're gonna use these methods anywhere else, please put them in a service, otherwise put them directly in the component